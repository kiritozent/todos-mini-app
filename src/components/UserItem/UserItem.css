.UserItem {
  background-color: #fff;
  cursor: pointer;
  transition: 0.3s;
  padding: 16px;
}

.UserNameText {
  font-size: medium;
  color: #3c4043;
  font-weight: 500;
}

.UserAvatarText {
  font-size: medium;
  color: #fff;
  font-weight: 500;
}

.UserEmailText {
  font-size: small;
}

.UserItem:hover {
  background-color: whitesmoke;
}
