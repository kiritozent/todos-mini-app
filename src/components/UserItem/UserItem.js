import React, { useState } from 'react';
import { List, Typography, Spin, Avatar } from 'antd';
import PropTypes from 'prop-types';
import './UserItem.css';
import randDarkColor from '../../helper/randDarkColor';

function UserItem(props) {
  const [avatarColor, setAvatarColor] = useState(randDarkColor());
  const { item, onClick, loading } = props;

  return (
    <List.Item className="UserItem" onClick={onClick}>
      <div
        style={{
          flexDirection: 'row',
          display: 'flex',
          justifyContent: 'center',
          alignItems: 'center',
        }}
      >
        <Avatar size="large" style={{ backgroundColor: avatarColor }}>
          <Typography.Text className="UserAvatarText">
            {item.name[0].toUpperCase()}
          </Typography.Text>
        </Avatar>
        <div
          style={{ flexDirection: 'column', display: 'flex', marginLeft: 16 }}
        >
          <Typography.Text className="UserNameText">
            {item.name}
          </Typography.Text>
          <Typography.Text>{item.email}</Typography.Text>
        </div>
      </div>
      {loading && <Spin size="small" />}
    </List.Item>
  );
}

UserItem.propTypes = {
  item: PropTypes.object.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default UserItem;
