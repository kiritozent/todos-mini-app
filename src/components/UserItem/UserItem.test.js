import React from 'react';
import { shallow } from 'enzyme';
import UserItem from './UserItem';

describe('UserItem', () => {
  it('render without crashing passed', () => {
    const testCase = shallow(<UserItem />);
    expect(testCase).toMatchSnapshot();
  });
});
