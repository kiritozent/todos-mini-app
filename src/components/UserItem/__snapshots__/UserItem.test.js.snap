// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserItem render without crashing passed 1`] = `
<Item
  className="UserItemContainer"
  onClick={[Function]}
>
  <div
    className="UserItem"
  >
    <Avatar
      shape="circle"
      size="large"
      style={
        Object {
          "backgroundColor": "#808080",
        }
      }
    >
      <Text
        className="UserAvatarText"
      />
    </Avatar>
    <div
      className="UserItemBody"
    >
      <Text
        className="UserNameText"
      />
      <Text />
    </div>
  </div>
</Item>
`;
